inference-engine/src/mkldnn_plugin/nodes/mkldnn_space_to_depth_node.h

```diff
@@ -15,25 +15,45 @@ class MKLDNNSpaceToDepthNode : public MKLDNNNode {
 public:
     MKLDNNSpaceToDepthNode(const std::shared_ptr<ngraph::Node>& op, const mkldnn::engine& eng, MKLDNNWeightsSharing::Ptr &cache);
 
+    static bool isSupportedOperation(const std::shared_ptr<const ngraph::Node>& op, std::string& errorMessage) noexcept;
     void getSupportedDescriptors() override;
     void initSupportedPrimitiveDescriptors() override;
     void createPrimitive() override;
     void execute(mkldnn::stream strm) override;
     bool created() const override;
 
-    static bool isSupportedOperation(const std::shared_ptr<const ngraph::Node>& op, std::string& errorMessage) noexcept;
+    void prepareParams() override;
+
+protected:
+    void executeDynamicImpl(mkldnn::stream strm) override;
 
 private:
     enum Mode {
         BLOCKS_FIRST = 0,
         DEPTH_FIRST = 1
     };
 
-    Mode mode;
-    size_t blockSize;
-    size_t blockStep;
-
-    std::unique_ptr<PermuteKernel> permuteKernel;
+    struct SpaceToDepthAttrs {
```

Comment:
please remove from structure attributes which can be changed in inference time
please see: https://github.com/openvinotoolkit/openvino/pull/8580